src/clinical/id-generator.ts

```diff
@@ -0,0 +1,293 @@
+/*
+ * Copyright (c) 2024 The Ontario Institute for Cancer Research. All rights reserved
+ *
+ * This program and the accompanying materials are made available under the terms of
+ * the GNU Affero General Public License v3.0. You should have received a copy of the
+ * GNU Affero General Public License along with this program.
+ *  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND ANY
+ * EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES
+ * OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT
+ * SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+ * INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED
+ * TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS;
+ * OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER
+ * IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+ * ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+import axios from 'axios';
+import memoize from 'memoizee';
+import ms from 'ms';
+import {
+	Biomarker,
+	ClinicalInfo,
+	Comorbidity,
+	CompletionStats,
+	Donor,
+	Exposure,
+	FamilyHistory,
+	FollowUp,
+	PrimaryDiagnosis,
+	SchemaMetadata,
+	Specimen,
+	Treatment,
+} from './clinical-entities';
+import { ClinicalEntitySchemaNames } from '../common-model/entities';
+import { config } from '../config';
+
+interface IdGenerationRequest {
+	programId: string;
+	submitterId: any;
+	submitterDonorId: string;
+	testInterval: string;
+	family_relative_id: string;
+	comorbidityTypeCode: string;
+	entityType: string;
+}
+
+const request = {
+	programId: 'null',
+	submitterId: 'null',
+	submitterDonorId: 'null',
+	testInterval: 'null',
+	family_relative_id: 'null',
+	comorbidityTypeCode: 'null',
+	entityType: 'null',
+};
+
+export interface PartialDonor {
```

Comment:
This basically duplicates type `Donor`, could we achieve the same thing doing `interface PartialDonor extends Partial<Donor> {}` or something similar?

Reply:
that's a good idea. Thanks.